src/app/components/search/search-error/search-error.component.ts

```diff
@@ -1,13 +1,30 @@
-import { Component, Input } from '@angular/core';
-import { NgSwitch, NgSwitchCase, NgSwitchDefault } from '@angular/common';
+import { NgIf, NgSwitch, NgSwitchCase, NgSwitchDefault } from '@angular/common';
+import { AfterViewInit, Component, Input, NgZone, inject } from '@angular/core';
+import { Router } from '@angular/router';
+import { InternalErrorComponent } from '@components/error/internal-server/internal-error.component';
+import { ErrorMessage } from '@darts-types/error-message.interface';
+import { HeaderService } from '@services/header/header.service';
 
 @Component({
   selector: 'app-search-error',
   standalone: true,
-  imports: [NgSwitchCase, NgSwitchDefault, NgSwitch],
+  imports: [NgSwitchCase, NgSwitchDefault, NgSwitch, InternalErrorComponent, NgIf],
   templateUrl: './search-error.component.html',
   styleUrls: ['./search-error.component.scss'],
 })
-export class SearchErrorComponent {
-  @Input() error: string | null = '';
+export class SearchErrorComponent implements AfterViewInit {
+  @Input() error!: ErrorMessage | null;
+
+  headerService = inject(HeaderService);
+  router = inject(Router);
+  private zone = inject(NgZone);
+
+  ngAfterViewInit(): void {
+    if (this.error?.display === 'PAGE' && this.error.status === 500) {
+      this.zone.run(() => {
```

Comment:
Why are we needing ngZone here?

Reply:
Functionality was working fine without but when running unit tests I was seeing "Navigation triggered outside Angular zone, did you forget to call 'ngZone.run()'" which was remediated after adding above